plone/app/z3cform/converters.py

```diff
@@ -264,7 +264,10 @@ def toFieldValue(self, value):
                                   for uid in value
                                   if uid in objects.keys())
         else:
-            return collectionType(v for v in value)
+            valueType = getattr(self.field.value_type, '_type', unicode)
+            if isinstance(valueType, tuple):
+                valueType = self.field.value_type[-1]
```

Comment:
You are correct, that was extraneous (only matters for collection types, non-sequitur here).  Corrected in 1e462cad1314548cf16eda7446d69b51d51c0a87 (also in PR).